### PR TITLE
Clean up JSON operator tokenizing code

### PR DESCRIPTION
**Pull request by alamb**
This has some small cleanups of the changes made in https://github.com/sqlparser-rs/sqlparser-rs/pull/913 by @izveigor

### CODE REVIEW

File: src/tokenizer.rs [hunk 1]
```diff
@@ -1086,8 +1079,8 @@ impl<'a> Tokenizer<'a> {
     }
 
     /// Tokenize an identifier or keyword, after the first char is already consumed.
-    fn tokenize_word(&self, first_chars: String, chars: &mut State) -> String {
-        let mut s = first_chars;
+    fn tokenize_word(&self, first_chars: impl Into<String>, chars: &mut State) -> String {
```
Comment (alamb) [inline src/tokenizer.rs:1082]: Uses some generics to avoid having to manually construct `String`s in several places.

File: src/tokenizer.rs [hunk 2]
```diff
@@ -934,19 +932,14 @@ impl<'a> Tokenizer<'a> {
                             match chars.peek() {
                                 Some(' ') => Ok(Some(Token::AtAt)),
                                 Some(tch) if self.dialect.is_identifier_start('@') => {
-                                    let mut s = ch.to_string();
```
Comment (alamb) [inline src/tokenizer.rs:937]: I think the new construction is both less verbose and clearer about intent

**Review by alamb — COMMENTED** [review]
cc @izveigor if you have a moment to review these changes I would appreciate it.

**Review by Dandandan — APPROVED** [review]
Nice cleanup

**Comment by alamb** [comment]
Thanks @Dandandan  and @izveigor

**Pull request merged**